src/Microsoft.ApiDesignGuidelines.Analyzers/Microsoft.ApiDesignGuidelines.Analyzers.md

```diff
@@ -170,6 +240,34 @@ Severity: Warning
 
 Help: [https://msdn.microsoft.com/en-us/library/bb264484.aspx](https://msdn.microsoft.com/en-us/library/bb264484.aspx)
 
+### CA1065: Do not raise exceptions in unexpected locations ###
+
+A method that is not expected to throw exceptions throws an exception.
```

Comment:
Comparing this description to the one for CA1050 above, I see that we're not being consistent in the type of information the description conveys. In this case (CA1065), the information is "Here's what you did that violated the rule". In CA1050, the information is "Here's why we have this rule". We need to choose one of these (or perhaps a third choice) so that rule descriptions have a uniform arrangement of their information content.